### PR TITLE
Add update indexer status

### PR DESCRIPTION
**Pull request by Jianzhuo**
Close issue [#68 Provide a helper function to update indexer status](https://github.com/subquery/network-contracts/issues/68)

### CODE REVIEW

File: contracts/RewardsHelper.sol [hunk 1]
```diff
@@ -54,6 +54,29 @@ contract RewardsHelper is Initializable, OwnableUpgradeable {
...
 
+    function updateIndexerStatus(address indexer) public {
+        RewardsDistributer rewardsDistributer = RewardsDistributer(settings.getRewardsDistributer());
+        IEraManager eraManager = IEraManager(settings.getEraManager());
+        uint256 currentEra =  eraManager.safeUpdateAndGetEra();
+        if(rewardsDistributer.getLastSettledEra(indexer) >= rewardsDistributer.getRewardInfo(indexer).lastClaimEra){
+            rewardsDistributer.collectAndDistributeRewards(indexer);
+        }
+        //apply all stakers' change of an indexer
+        while(rewardsDistributer.getPendingStakeChangeLength(indexer) > 0){
```
Comment (HZHAIX) [inline contracts/RewardsHelper.sol:65]: Use `getPendingStakers` to get all the stakers
User `batchApplyStakeChange` to apply the stakers change

File: contracts/RewardsHelper.sol [hunk 2]
```diff
@@ -54,6 +54,29 @@ contract RewardsHelper is Initializable, OwnableUpgradeable {
         }
     }
 
+    function updateIndexerStatus(address indexer) public {
+        RewardsDistributer rewardsDistributer = RewardsDistributer(settings.getRewardsDistributer());
+        IEraManager eraManager = IEraManager(settings.getEraManager());
+        uint256 currentEra =  eraManager.safeUpdateAndGetEra();
+        if(rewardsDistributer.getLastSettledEra(indexer) >= rewardsDistributer.getRewardInfo(indexer).lastClaimEra){
```
Comment (mzxyz) [inline contracts/RewardsHelper.sol]: Check `ewardsDistributer.getRewardInfo(indexer).lastClaimEra < currentEra - 1`

File: contracts/RewardsHelper.sol [hunk 3]
```diff
@@ -54,6 +54,29 @@ contract RewardsHelper is Initializable, OwnableUpgradeable {
...
+            rewardsDistributer.collectAndDistributeRewards(indexer);
+        }
+        //apply all stakers' change of an indexer
+        while(rewardsDistributer.getPendingStakeChangeLength(indexer) > 0){
+            address staker = rewardsDistributer.getPendingStaker(indexer, rewardsDistributer.getPendingStakeChangeLength(indexer) - 1);
+            rewardsDistributer.applyStakeChange(indexer, staker);
+        }
+        //apply indexer's commission rate change
+        if(rewardsDistributer.getCommissionRateChangedEra(indexer) != 0 && rewardsDistributer.getCommissionRateChangedEra(indexer) <= currentEra){
```
Comment (mzxyz) [inline contracts/RewardsHelper.sol]: uint256 ICREra = rewardsDistributer.getCommissionRateChangedEra(indexer)
if (ICREra != 0 && ICREra <=currentEra ) {...}

File: test/RewardsDistributer.test.ts [hunk 4]
```diff
@@ -207,6 +207,77 @@ describe('RewardsDistributer Contract', () => {
...
+            await rewardsHelper.updateIndexerStatus(indexer.address);
+            await staking.connect(delegator).delegate(indexer.address, etherParse('1'));
+            await staking.connect(delegator2).delegate(indexer.address, etherParse('2'));
+            await staking.connect(indexer).setCommissionRate(200);
+            await startNewEra(mockProvider, eraManager);
+            await startNewEra(mockProvider, eraManager);
+            await startNewEra(mockProvider, eraManager);
+            await rewardsHelper.updateIndexerStatus(indexer.address);
+            expect(await eraManager.eraNumber()).to.be.eq(8);
+            expect((await rewardsDistributor.getRewardInfo(indexer.address)).lastClaimEra).to.be.eq(7);
```
Comment (mzxyz) [inline test/RewardsDistributer.test.ts:242]: - Check `lastSettleEra` as well
- pendingStakers.length === 0
- currentCommission = changed value

**Comment by mzxyz** [comment]
One more test case for indexer who is:

```
- have delegator
- unregistered
- call updateStatus
- delegator can undelegate from the unregistered indexer
```

**Pull request merged**